scripts/create-package/plopfile.ts

```diff
@@ -163,35 +183,63 @@ module.exports = (plop: NodePlopAPI) => {
 
-  // Copy dep versions in package.json from actual current versions.
-  // This is preferable over hardcoding dependency versions to keep things in sync.
-  // As of writing, @fluentui/react-experiments also depends on all the packages the React template needs,
-  // so we grab the current versions from there (or @fluentui/codemods for the node template).
-  const referencePackage = target === 'node' ? 'codemods' : 'react-experiments';
-  const referencePackageJson: PackageJson = fs.readJSONSync(
-    path.join(root, 'packages', referencePackage, 'package.json'),
+  // Read the package.json files of the given reference packages and combine into one object.
+  // This way if a dep is defined in any of them, it can easily be copied to newPackageJson.
+  const referenceDeps: Pick<PackageJson, 'dependencies' | 'devDependencies' | 'peerDependencies'> = _.merge(
+    {},
+    ...referencePackages.map(pkg =>
+      _.pick(fs.readJSONSync(path.join(root, 'packages', pkg, 'package.json')), ...depTypes),
+    ),
   );
-  const newPackageJson: PackageJson = JSON.parse(packageJsonContents);
+  console.dir(referenceDeps);
+
+  let hasError = false;
 
-  for (const depType of ['dependencies', 'devDependencies', 'peerDependencies']) {
+  for (const depType of depTypes) {
     if (!newPackageJson[depType]) {
       continue;
     }
-    for (const pkg of Object.keys(newPackageJson[depType])) {
-      if (!hasTests && /\b(jest|enzyme|test)\b/.test(pkg)) {
-        delete newPackageJson[depType][pkg];
-      } else if (referencePackageJson[depType]?.[pkg]) {
-        newPackageJson[depType][pkg] = referencePackageJson[depType][pkg];
+    for (const depPkg of Object.keys(newPackageJson[depType])) {
+      if (!hasTests && /\b(jest|enzyme|test|react-conformance)\b/.test(depPkg)) {
+        delete newPackageJson[depType][depPkg];
+      } else if (referenceDeps[depType]?.[depPkg]) {
+        newPackageJson[depType][depPkg] = referenceDeps[depType][depPkg];
       } else {
+        hasError = true;
         console.warn(
-          `Couldn't determine appropriate ${depType} version of ${pkg} from ${referencePackage} package.json`,
+          chalk.yellow(
+            `Couldn't determine appropriate ${depType} version of ${depPkg} from ${referencePackages.join(
+              ' or ',
+            )} package.json`,
+          ),
         );
       }
     }
   }
+  return hasError;
+}
+
+function updatePackageJson(packageJsonContents: string, answers: Answers) {
+  const { target, hasTests, publish } = answers;
+
+  // Copy dep versions in package.json from actual current version specs.
+  // This is preferable over hardcoding dependency versions to keep things in sync.
+  // The reference package(s) may need to be updated over time as dependency lists change.
+  const newPackageJson: PackageJson = JSON.parse(packageJsonContents);
+  const referencePackages = target === 'node' ? ['codemods'] : ['react-button', 'react-image'];
```

Comment:
What do you think about explicitly setting dependencies? Imo we should be aware of the minimum requirements for a package on our repo.

Doest block this fix though

Reply:
What do you mean by explicitly setting dependencies? It would definitely be possible to (for example) get the actual current version of each package from its package.json (then prepend a `^`), but when I implemented this originally I thought that pulling the complete semver specs from an existing package would produce better consistency.

Reply:
Sometime in the past we had specific versions, but those constantly got out of sync, and then caused breaks in syncpack once we enabled that.

Reply:
I think I would prefer a new package to not start with anything other than ^ notation, it should be a choice of the package author to do otherwise since generally there is not a good reason not to.

We should also not rely on references packages and make it clear what min requirements should be for a node/react package

Reply:
I can see the argument for choice of package author in some ways, but I still think we should keep the current behavior as the default. All these dependencies are used in many places across the repo and need to have the same semver spec (which in some cases must be a specific version--for example react and its types) everywhere to satisfy the syncpack check. And the reason we added that check was to prevent packages from accidentally getting out of sync (because in almost all cases that's not intentional) and installing duplicates. In the case that the package author *does* have a specific reason they need a different dep version, they can manually change it and add an override to the syncpack config.